src/Product/SearchProvider.php

```diff
@@ -452,15 +453,29 @@ private function hideUselessFacets(array $facets)
                 continue;
             }
 
+            $totalFacetProducts = 0;
             $usefulFiltersCount = 0;
             foreach ($facet->getFilters() as $filter) {
                 if ($filter->getMagnitude() > 0) {
+                    $totalFacetProducts += $filter->getMagnitude();
                     ++$usefulFiltersCount;
                 }
             }
 
             $facet->setDisplayed(
+                // There are two filters displayed
                 $usefulFiltersCount > 1
+                ||
+                (
+                    /*
+                     * There is only one fitler and the
+                     * magnitude is different than the
+                     * total products
+                     */
+                    count($facet->getFilters()) === 1
+                    && $totalFacetProducts != $totalProducts
+                    && $usefulFiltersCount > 0
```

Comment:
That's not the easier code to read. Maybe having the 2 comments at the same indentation would help?

Reply:
something like.
```php
                    count($facet->getFilters()) === 1 // There is only one filter
                    && $totalFacetProducts != $totalProducts // And the magnitude is different
```

Reply:
```php
            $facet->setDisplayed(
                // There are two filters displayed
                $usefulFiltersCount > 1
                ||
                /*
                 * There is only one fitler and the
                 * magnitude is different than the
                 * total products
                 */
                (
                    count($facet->getFilters()) === 1
                    && $totalFacetProducts != $totalProducts
                    && $usefulFiltersCount > 0
                )
            );

Reply:
In all cases, this test seems complex to read because of the comment in the middle of the statements  :)

Reply:
Thanks